### PR TITLE
Auto generate list of subprojects & working-groups from annual report year

### PR DESCRIPTION
**Pull request by Priyankasaggu11929**
The PR implements auto generating the list of "New", "Retired" and "Continuing" sub-projects & reporting working-groups under a certain SIG, in the `annual-report/sig_report.tmpl` go template file.

**Which issue(s) this PR fixes**:

Part of https://github.com/kubernetes/community/issues/6357

---

One of the examples for the filled "Subprojects" & "Working groups" section (in case - "sig-node") looks like following:

Screenshot 2023-01-20 at 10 14 16 PM

### CODE REVIEW

**Comment by Priyankasaggu11929** [comment]
/retitle Auto generate list of subprojects & working-groups from annual report year

File: generator/annual-report/sig_report.tmpl [hunk 1]
```diff
@@ -71,43 +71,56 @@
...
--
+## [Subprojects](https://git.k8s.io/community/{{.Dir}}#subprojects)
+{{ define "subprojects" -}}
+{{- $owingsig := .Dir -}}
+{{- if .New}}
+**New in {{lastYear}}:**
+{{ range .New}}
+  - {{.}}
+{{ end}}
+{{- end}}
```
Comment (cblecker) [inline generator/annual-report/sig_report.tmpl:82]:
```suggestion
## [Subprojects](https://git.k8s.io/community/{{.Dir}}#subprojects)
{{- define "subprojects" -}}
{{- $owingsig := .Dir -}}
{{- if .New}}
**New in {{lastYear}}:**
{{range .New}}
  - {{.}}
{{- end}}
{{- end}}
```

File: generator/annual-report/sig_report.tmpl [hunk 2]
```diff
@@ -71,43 +71,56 @@
...
 
-Retired in $YYYY:
-- [$subproject-name](https://git.k8s.io/community/$sig-id#$subproject-name)
--
+{{- if .Retired}}
+**Retired in {{lastYear}}:**
+{{ range .Retired}}
+  - {{.}}
+{{ end}}
+{{- end}}
```
Comment (cblecker) [inline generator/annual-report/sig_report.tmpl:89]:
```suggestion
{{- if .Retired}}

**Retired in {{lastYear}}:**
{{range .Retired}}
  - {{.}}
{{- end}}
{{- end}}
```

File: generator/annual-report/sig_report.tmpl [hunk 3]
```diff
@@ -71,43 +71,56 @@
...
-Continuing:
-- [$subproject-name](https://git.k8s.io/community/$sig-id#$subproject-name)
--
+{{- if .Continuing}}
+**Continuing:**
+{{ range .Continuing}}
+  - {{.}}
+{{ end}}
+{{- end}}
+{{- end -}}
```
Comment (cblecker) [inline generator/annual-report/sig_report.tmpl:97]:
```suggestion
{{- if .Continuing}}

**Continuing:**
{{range .Continuing}}
  - {{.}}
{{- end}}
{{- end}}
{{- end}}
```

File: generator/annual-report/sig_report.tmpl [hunk 4]
```diff
@@ -71,43 +71,56 @@
...
-- [$wg-name](https://git.k8s.io/community/$wg-id/) ([$YYYY report](https://git.k8s.io/community/$wg-id/annual-report-$YYYY.md))
--
+## [Working groups](https://git.k8s.io/community/{{.Dir}}#working-groups)
+{{ $categorizedWorkingGroups := getCategorizedWorkingGroups .Dir }}
+{{- if $categorizedWorkingGroups.New}}
+**New in {{lastYear}}:**
+{{ range $categorizedWorkingGroups.New }}
+ - {{.}}
+{{end}}
+{{- end}}
```
Comment (cblecker) [inline generator/annual-report/sig_report.tmpl:109]:
```suggestion
## [Working groups](https://git.k8s.io/community/{{.Dir}}#working-groups)
{{ $categorizedWorkingGroups := getCategorizedWorkingGroups .Dir }}
{{- if $categorizedWorkingGroups.New}}

**New in {{lastYear}}:**
{{range $categorizedWorkingGroups.New }}
 - {{.}}
{{- end}}
{{- end}}
```

File: generator/annual-report/sig_report.tmpl [hunk 5]
```diff
@@ -71,43 +71,56 @@
...
 
-Retired in $YYYY:
-- [$wg-name](https://git.k8s.io/community/$wg-id/) ([$YYYY report](https://git.k8s.io/community/$wg-id/annual-report-$YYYY.md))
--
+{{- if $categorizedWorkingGroups.Retired}}
+**Retired in {{lastYear}}:**
+{{ range $categorizedWorkingGroups.Retired }}
+ - {{.}}
+{{end}}
+{{- end}}
```
Comment (cblecker) [inline generator/annual-report/sig_report.tmpl:116]:
```suggestion
{{- if $categorizedWorkingGroups.Retired}}

**Retired in {{lastYear}}:**
{{range $categorizedWorkingGroups.Retired }}
 - {{.}}
{{- end}}
{{- end}}
```

File: generator/annual-report/sig_report.tmpl [hunk 6]
```diff
@@ -71,43 +71,56 @@
...
 
-Continuing:
-- [$wg-name](https://git.k8s.io/community/$wg-id/) ([$YYYY report](https://git.k8s.io/community/$wg-id/annual-report-$YYYY.md))
--
+{{- if $categorizedWorkingGroups.Continuing}}
+**Continuing:**
+{{ range $categorizedWorkingGroups.Continuing }}
+ - {{ . }}
+{{end}}
+{{- end}}
```
Comment (cblecker) [inline generator/annual-report/sig_report.tmpl:123]:
```suggestion
{{- if $categorizedWorkingGroups.Continuing}}

**Continuing:**
{{ range $categorizedWorkingGroups.Continuing }}
 - {{ . }}
{{- end}}
{{- end}}
```

**Review by cblecker — COMMENTED** [review]
Formatting changes to output

**Comment by cblecker** [comment]
Wow @Priyankasaggu11929! Other than a few formatting nits, this is amazing work. Thanks for this!

**Comment by Priyankasaggu11929** [comment]
Thanks @cblecker. I've updated the PR with the suggested formatting changes. PTAL. :)

**Comment by nikhita** [comment]
Retroactive lgtm :) 

Thanks for automating this!

**Pull request merged**